kythe/go/platform/tools/kzip_validator/kzip_validator.go

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Binary kzip_validator checks the contents of a kzip against a code repo and
+// compares file coverage.
+//
+// Example:
+//  kzip_validator -kzip <kzip-file> -local_repo <repo-root-dir> -lang cc,h
+//  kzip_validator -kzip <kzip-file> -repo_url <repo-url> -lang java [-version <hash>]
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"kythe.io/kythe/go/extractors/validation"
+
+	"bitbucket.org/creachadair/stringset"
+)
+
+var (
+	kzip      = flag.String("kzip", "", "The comma separated list of kzip files to check")
+	repoURL   = flag.String("repo_url", "", "The repo to compare against")
+	version   = flag.String("version", "", "The version of the remote repo to compare")
+	localRepo = flag.String("local_repo", "", "The path of an optional local repo to specify instead of -repoURL")
+	lang      = flag.String("lang", "", "The comma-separated language file extensions to check, e.g. 'java' or 'cpp,h'")
+
+	missingFile = flag.String("missing_file", "", "An optional file to write all missing filepaths to")
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `Usage: %s -kzip <kzip-file> -repoURL <url> [-version <hash>]
+
+Compare a kzip file's contents with a given repo, and print results of file coverage.
+
+Options:
+`, filepath.Base(os.Args[0]))
+		flag.PrintDefaults()
+	}
+}
+
+func verifyFlags() {
+	if len(flag.Args()) > 0 {
+		log.Fatalf("Unknown arguments: %v", flag.Args())
+	}
+
+	if *kzip == "" {
+		log.Fatal("You must provide at least one -kzip file")
+	}
+	if *repoURL == "" && *localRepo == "" {
+		log.Fatal("You must specify either -repo_url or a -local_repo path")
+	}
+	if *repoURL != "" && *localRepo != "" {
+		log.Fatal("You must specify only one of -repo_url or -local_repo")
+	}
+	if *version != "" && *repoURL == "" {
+		log.Fatal("-version is only copmatible with -reop_url, not -local_repo")
+	}
+	if *lang == "" {
+		log.Fatal("You must specify what -lang file extensions to examine")
+	}
+}
+
+func main() {
+	flag.Parse()
+	verifyFlags()
+
+	fmt.Printf("Validating kzip %s\n", *kzip)
+
+	repoPath := fetchRepo()
+
+	s := validation.Settings{
+		Compilations: strings.Split(*kzip, ","),
+		Repo:         repoPath,
+		Langs:        stringset.FromKeys(strings.Split(*lang, ",")),
+	}
+	if *missingFile != "" {
```

Comment:
If empty-string is the "do nothing" case anyway, this can more simply be part of the initializer above:

```go
   MissingOutput: *missingFile,
```

Reply:
done